packages/ec-core/src/distributions/collection.rs

```diff
@@ -1,92 +1,194 @@
+/// length.
 ///
-/// The `CollectionGenerator` struct is used to generate a collection of random
-/// elements. It takes an element generator and a size as input, and generates a
-/// `Vec` of random elements based on the specified size and the mechanism for
-/// generating random elements.
+/// Sampling from this distribution samples from the element collection `length`
+/// times.
 ///
-/// The module also defines the `ConvertToCollectionGenerator` trait, which
-/// provides methods for converting a type into a `CollectionGenerator`. This
-/// trait is implemented for any type that implements the `Generator` trait.
+/// # Provided conversion methods
 ///
-/// Finally, the module implements the `Generator` trait for the
-/// `CollectionGenerator` struct, allowing it to generate a `Vec` of random
-/// elements using the `generate` method.
-use rand::prelude::Distribution;
-
-/// Information for generating a collection of random elements.
+/// For ease of use, the trait [`ConvertToCollectionDistribution`] provides
+/// chainable constructors on any [`Distribution`] to turn that distribution
+/// into a collection distribution.
+///
+/// Those constructors are:
+/// - [`ConvertToCollectionDistribution::into_collection`], a owning
+///   constructor, moving the child distribution
+/// - [`ConvertToCollectionDistribution::to_collection`], a borrowing
+///   constructor, borrowing the child distribution.
+///
+/// Usage:
+///
+/// ```
+/// # use rand::distr::StandardUniform;
+/// # use ec_core::distributions::collection::{Collection, ConvertToCollectionDistribution};
+/// #
+/// let collection_distribution: Collection<StandardUniform> = StandardUniform.into_collection(10);
+/// # let _ = collection_distribution;
+/// ```
+///
+/// ```
+/// # use rand::distr::StandardUniform;
+/// # use ec_core::distributions::collection::{Collection, ConvertToCollectionDistribution};
+/// #
+/// let collection_distribution: Collection<&StandardUniform> = StandardUniform.to_collection(10);
+/// # let _ = collection_distribution;
+/// ```
 ///
-/// `size` indicates how many elements to generate.
-/// `element_generator` is used to generate individual elements.
-pub struct Generator<C> {
-    pub element_generator: C,
-    pub size: usize,
+/// Also see [`ConvertToCollectionDistribution`]
+///
+/// # Example
+/// ```
+/// # use rand::{distr::{Distribution, StandardUniform}, rng};
+/// # use ec_core::distributions::collection::Collection;
+/// #
+/// let singular_distribution = StandardUniform;
+/// let collection_distribution = Collection::new(&singular_distribution, 10);
+///
+/// let mut rng = rng();
+///
+/// let single_element: i32 = singular_distribution.sample(&mut rng);
+/// # let _ = single_element;
+/// let collection: Vec<i32> = collection_distribution.sample(&mut rng);
+/// assert_eq!(collection.len(), 10)
+/// ```
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Default)]
+pub struct Collection<C> {
```

Comment:
Either move the `Generator` -> `Collection` rename into its own PR or join it with the `Distribution` renamings.